test/crowdfund/AtomicManualParty.t.sol

```diff
@@ -0,0 +1,161 @@
+    }
+
+    function test_createAtomicManualPartyWithMetadata() public {
+        Party.PartyOptions memory opts;
+        opts.name = "PARTY";
+        opts.symbol = "PR-T";
+        opts.governance.voteDuration = 99;
+        opts.governance.executionDelay = _EXECUTION_DELAY;
+        opts.governance.passThresholdBps = 1000;
+        opts.governance.totalVotingPower = 180;
+
+        address[] memory partyMembers = new address[](2);
+        uint96[] memory partyMemberVotes = new uint96[](2);
+
+        partyMembers[0] = john;
+        partyMembers[1] = danny;
+
+        partyMemberVotes[0] = 100;
+        partyMemberVotes[1] = 80;
+
+        // Not checking address of the party
+        vm.expectEmit(false, true, true, true);
+        emit PartyCreated(
+            Party(payable(0)),
+            opts,
+            preciousTokens,
+            preciousTokenIds,
+            address(atomicManualParty)
+        );
+        vm.expectEmit(false, true, true, true);
+        emit ProviderSet(address(0), IMetadataProvider(address(0)));
+        Party atomicParty = atomicManualParty.createPartyWithMetadata(
+            Party(payable(address(Proxy(payable(address(party))).IMPL()))),
+            opts,
+            preciousTokens,
+            preciousTokenIds,
+            0,
+            MetadataProvider(address(0)),
+            "",
+            partyMembers,
+            partyMemberVotes
+        );
+
+        // Ensure `atomicManualParty` is not an authority after creation
+        assertFalse(party.isAuthority(address(atomicManualParty)));
+
+        // Ensure holders match input
+        vm.roll(block.number + 1);
+        vm.warp(block.timestamp + 1);
+        assertEq(atomicParty.getVotingPowerAt(john, uint40(block.timestamp)), 100);
+        assertEq(atomicParty.getVotingPowerAt(danny, uint40(block.timestamp)), 80);
+    }
+
+    function test_createAtomicManualPartyArityMismatch() public {
+        Party.PartyOptions memory opts;
+        opts.name = "PARTY";
+        opts.symbol = "PR-T";
+        opts.governance.voteDuration = 99;
+        opts.governance.executionDelay = _EXECUTION_DELAY;
+        opts.governance.passThresholdBps = 1000;
+        opts.governance.totalVotingPower = 180;
+
+        address[] memory partyMembers = new address[](3);
+        uint96[] memory partyMemberVotes = new uint96[](2);
+
+        partyMembers[0] = john;
+        partyMembers[1] = danny;
+        partyMembers[2] = steve;
+
+        partyMemberVotes[0] = 100;
+        partyMemberVotes[1] = 80;
+
+        Party partyImpl = Party(payable(address(Proxy(payable(address(party))).IMPL())));
+        vm.expectRevert(AtomicManualParty.PartyMembersArityMismatch.selector);
+        atomicManualParty.createParty(
+            partyImpl,
+            opts,
+            preciousTokens,
+            preciousTokenIds,
+            0,
+            partyMembers,
+            partyMemberVotes
+        );
+    }
+}
```

Comment:
Add a testcase for when `createParty` is called with multiple cards minted to same address?